cmd/kube-rbac-proxy/app/options/proxyoptions.go

```diff
@@ -125,13 +139,101 @@ func (o *ProxyOptions) ApplyTo(c *server.KubeRBACProxyInfo, a *serverconfig.Auth
 		}
 	}
 
+	c.UpstreamHeaders = o.UpstreamHeader
 	c.IgnorePaths = o.IgnorePaths
 	c.AllowPaths = o.AllowPaths
 	a.APIAudiences = o.TokenAudiences
 
 	return nil
 }
 
+func validateSecureConnectionConfig(o *ProxyOptions) error {
+	if !identityheaders.HasIdentityHeadersEnabled(o.UpstreamHeader) || !o.UpstreamForceH2C {
+		return nil
+	}
+
+	ctx, cancel := context.WithTimeout(context.Background(), time.Duration(loopbackLookupTimeout)*time.Second)
+	defer cancel()
+
+	isLoopback, errLoopback := isLoopbackAddress(ctx, o.Upstream)
```

Comment:
move inside of the `if o.UpstreamForceH2C` branch?

Reply:
This is necessary for identity headers too.